.github/workflows/clang-format-checker.yml

```diff
@@ -84,23 +85,24 @@ jobs:
         uses: actions/checkout@v4
         with:
           fetch-depth: 2
+          path: main
```

Comment:
Should this really be `main` under the source root of the tree?  Won't that cause problems too?  Could this be changed to be under a `.gitignore` directory, such as `build`?  Like `build/main_src`?  Also, how does this action select the branch to check out?  Is it just defaulting to `main`?  I admit I'm not that familiar with the command structure here, but there appears to be some magic/defaults that don't appear explicitly defined in this file.  Like where is the PR branch placed?  Is it placing us at the source root of that checkout by default?

Reply:
build/main_src should be fine.
I'll update it.
The default for actions/checkout@v4 should be current repo with default branch.
And you can change it by set ref and repository.